app/services/action_builder.rb

```diff
@@ -0,0 +1,23 @@
+module ActionBuilder
```

Comment:
Turned this into a mixin which allows sharing the logic between the classes that need it. Do we need to spec this now?